apps/vs-code-react/src/app/dataMapper/appV2.tsx

```diff
@@ -1,3 +1,207 @@
+import type { AppDispatch, RootState } from '../../state/store';
+import { VSCodeContext } from '../../webviewCommunication';
+import { DataMapperFileService} from './services/dataMapperFileService';
+import {
+  getFileNameAndPath,
+  DataMapDataProvider,
+  DataMapperDesigner,
+  DataMapperDesignerProvider,
+  InitDataMapperApiService,
+  defaultDataMapperApiServiceOptions,
+  getFunctions,
+  getSelectedSchema,
+} from '@microsoft/logic-apps-data-mapper-v2';
+import { getTheme, useThemeObserver } from '@microsoft/logic-apps-designer';
+import type { Theme } from '@microsoft/logic-apps-shared';
+import type { MessageToVsix } from '@microsoft/vscode-extension-logic-apps';
+import { ExtensionCommand } from '@microsoft/vscode-extension-logic-apps';
+import { useCallback, useContext, useEffect, useMemo, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+// interface SchemaFile {
+//   path: string;
+//   type: SchemaType;
+// }
+
 export const DataMapperAppV2 = () => {
-  return <></>;
+  const dispatch = useDispatch<AppDispatch>();
+  const vscode = useContext(VSCodeContext);
+  const [theme, setTheme] = useState<Theme>(getTheme(document.body));
+  const xsltFilename = useSelector((state: RootState) => state.dataMapDataLoader.xsltFilename);
+  const xsltContent = useSelector((state: RootState) => state.dataMapDataLoader.xsltContent);
+  const mapDefinition = useSelector((state: RootState) => state.dataMapDataLoader.mapDefinition);
+  const mapMetadata = useSelector((state: RootState) => state.dataMapDataLoader.dataMapMetadata);
+  const sourceSchemaFilename = useSelector((state: RootState) => state.dataMapDataLoader.sourceSchemaFilename);
+  const sourceSchema = useSelector((state: RootState) => state.dataMapDataLoader.sourceSchema);
+  const targetSchemaFilename = useSelector((state: RootState) => state.dataMapDataLoader.targetSchemaFilename);
+  const targetSchema = useSelector((state: RootState) => state.dataMapDataLoader.targetSchema);
+  // const schemaFileList = useSelector((state: RootState) => state.dataMapDataLoader.schemaFileList);
+  const customXsltPathsList = useSelector((state: RootState) => state.dataMapDataLoader.customXsltPathsList);
+  const fetchedFunctions = useSelector((state: RootState) => state.dataMapDataLoader.fetchedFunctions);
+
+  const runtimePort = useSelector((state: RootState) => state.dataMapDataLoader.runtimePort);
+
+  if (runtimePort) {
+    InitDataMapperApiService({
+      ...defaultDataMapperApiServiceOptions,
+      port: runtimePort ?? defaultDataMapperApiServiceOptions.port,
+    });
+  }
+
+  const sendMsgToVsix = useCallback(
+    (msg: MessageToVsix) => {
+      vscode.postMessage(msg);
+    },
+    [vscode]
+  );
+
+  const dataMapperFileService = useMemo(() => {
+    return new DataMapperFileService(sendMsgToVsix);
+  } , [sendMsgToVsix]);
+
+  // const addSchemaFromFile = (selectedSchemaFile: SchemaFile) => {
```

Comment:
can we please remove all the commented-out code from the PR? We can use this PR if we need to reference that code again?

Reply:
I feel like we're already missing out on code that we already have functionality for, so idk about this. Let's discuss during our dev sync?